### PR TITLE
MAINT: use sysconfig not distutils.sysconfig where possible

### PR DESCRIPTION
**Pull request by mattip**
Since python 3.2 `sysconfig` has moved to the stdlib. Some functionality is still needed from `distutils.sysconfig`, for instance `distutils.sysconfig.customize_compiler` and `distutils.sysconfig.get_python.lib`.

It is not clear if this is a (very small) step toward fixing the distutils/setuptools mess, but it seems like the right thing to do anyway.

### CODE REVIEW

File: numpy/distutils/system_info.py [hunk 1]
```diff
@@ -187,6 +187,7 @@
 import tempfile
 import shutil
 
+__all__ = ['system_info']
```
Comment (mattip) [inline numpy/distutils/system_info.py:190]: This file was missing an `__all__`, so it was re-exporting imported modules.

File: numpy/distutils/unixccompiler.py [hunk 2]
```diff
@@ -26,7 +26,8 @@ def UnixCCompiler__compile(self, obj, src, ext, cc_args, extra_postargs, pp_opts
         self.compiler_so = ccomp
     # ensure OPT environment variable is read
     if 'OPT' in os.environ:
-        from distutils.sysconfig import get_config_vars
+        # XXX who uses this?
+        from sysconfig import get_config_vars
```
Comment (mattip) [inline numpy/distutils/unixccompiler.py:30]: When is `OPT` ever used?

File: numpy/tests/test_public_api.py [hunk 3]
```diff
@@ -367,18 +367,6 @@ def test_all_modules_are_expected():
...
-    'numpy.distutils.system_info.log',
-    'numpy.distutils.system_info.os',
-    'numpy.distutils.system_info.platform',
-    'numpy.distutils.system_info.re',
-    'numpy.distutils.system_info.shutil',
-    'numpy.distutils.system_info.subprocess',
-    'numpy.distutils.system_info.sys',
-    'numpy.distutils.system_info.tempfile',
-    'numpy.distutils.system_info.textwrap',
-    'numpy.distutils.system_info.warnings',
```
Comment (mattip) [inline numpy/tests/test_public_api.py:381]: Now that `system_info` has an `__all__`, these can be removed.

File: runtests.py [hunk 4]
```diff
@@ -445,7 +445,7 @@ def build_project(args):
         os.makedirs(site_dir)
     if not os.path.exists(site_dir_noarch):
         os.makedirs(site_dir_noarch)
-    env['PYTHONPATH'] = site_dir + ':' + site_dir_noarch
+    env['PYTHONPATH'] = site_dir + os.pathsep + site_dir_noarch
```
Comment (mattip) [inline runtests.py:448]: This mistake makes me suspicious - do we really need to set `PYTHONPATH` here? It is done correctly before running tests.

**Comment by eric-wieser** [comment]
For reference: [`distutils.sysconfig`](https://docs.python.org/3/distutils/apiref.html#module-distutils.sysconfig), [`sysconfig`](https://docs.python.org/3/library/sysconfig.html)

File: numpy/distutils/system_info.py [hunk 5]
```diff
@@ -255,7 +256,7 @@ def libpaths(paths, bits):
 
 if sys.platform == 'win32':
     default_lib_dirs = ['C:\\',
-                        os.path.join(distutils.sysconfig.EXEC_PREFIX,
+                        os.path.join(sysconfig.EXEC_PREFIX,
```
Comment (eric-wieser) [inline numpy/distutils/system_info.py:259]: AttributeError: module 'sysconfig' has no attribute 'EXEC_PREFIX'

**Comment by mattip** [comment]
So this is now passing, and could be committed as-is. There is more to do:
- `distutils.sysconfig.customize_compiler` has no counterpart in `sysconfig`, it probably should be a method of `ccompiler` clas.s
- I think `distutils.sysconfig.get_python_lib(standard_lib=1)` could be replaced by `sysconfig.get_path('data')` but was not sure enough to put it into this PR
- `distutils.sysconfig.get_config_var('EXEC_PREFIX') and sysconfig.get_config_var('EXEC_PREFIX') return different values on windows, is this a bug or intended behavior?

**Comment by mattip** [comment]
xref python/cpython#22088

**Comment by charris** [comment]
Thanks Matti.

**Pull request merged**